src/test/java/ca/corefacility/bioinformatics/irida/service/impl/unit/user/UserServiceImplTest.java

```diff
@@ -128,6 +130,20 @@ public void testCreateGoodPassword() {
 		verify(userRepository).save(u);
 	}
 
+	@Test
+	public void testReadAnnouncements() {
+		final User u = user();
+
+		List<Announcement> readAnnouncements = Lists.newArrayList(new Announcement("test 1", "this is a test message", true, u),
+				new Announcement("test 2", "this is also a test message", false, u));
+
+		doReturn(readAnnouncements).when(announcementUserJoinRepository).getAnnouncementsReadByUser(u);
```

Comment:
I don't think I've seen `doReturn` used before.  Is `doReturn(x).when(y)` the same as `when(x).thenReturn(y)`?  If so can you switch it to just use the `when x then y` version?  That's what we've used in the rest of the project.

Reply:
Please see 6099bdcc29ce4d455658929e3f8e486473a0ccac.

Reply:
Can we actually beef this test up a little bit so that it _does_ have announcements to mark as read when a user is created?  Or else add a second test with that functionality?  I know we have one in the ITs, but one here would be good too.